pkg/provisioner/ironic/testserver/redfish.go

```diff
@@ -0,0 +1,703 @@
+	return &RedfishMock{
+		MockServer: NewTLS(t, "redfish"),
+	}
+}
+
+// Endpoint returns the URL to the server
+func (m *RedfishMock) Endpoint() string {
+	return m.MockServer.RawEndpoint()
+}
+
+// RootService configures the server with a valid response for /redfish/v1/
+func (m *RedfishMock) rootService() *RedfishMock {
+	m.ResponseWithCode("/redfish/v1/", `
+	{
+		"@odata.type": "#ServiceRoot.v1_9_0.ServiceRoot",
+		"Id": "RootService",
+		"Name": "Root Service",
+		"RedfishVersion": "1.6.0",
+		"UUID": "92384634-2938-2342-8820-489239905423",
+		"Systems": {
+			"@odata.id": "/redfish/v1/Systems"
+		},
+		"Chassis": {
+			"@odata.id": "/redfish/v1/Chassis"
+		},
+		"Managers": {
+			"@odata.id": "/redfish/v1/Managers"
+		},
+		"Tasks": {
+			"@odata.id": "/redfish/v1/TaskService"
+		},
+		"SessionService": {
+			"@odata.id": "/redfish/v1/SessionService"
+		},
+		"AccountService": {
+			"@odata.id": "/redfish/v1/AccountService"
+		},
+		"EventService": {
+			"@odata.id": "/redfish/v1/EventService"
+		},
+		"Registries": {
+			"@odata.id": "/redfish/v1/Registries"
+		},
+		"UpdateService": {
+			"@odata.id": "/redfish/v1/UpdateService"
+		},
+		"CertificateService": {
+			"@odata.id": "/redfish/v1/CertificateService"
+		},
+		"Links": {
+			"Sessions": {
+				"@odata.id": "/redfish/v1/SessionService/Sessions"
+			}
+		},
+		"Oem": { },
+		"@odata.id": "redfish/v1/"
+	}`, http.StatusOK)
+	return m
+}
+
+// Unauthorized configures the server with a http 401 Unauthorized response for /redfish/v1/
+func (m *RedfishMock) Unauthorized() *RedfishMock {
+	m.ErrorResponse("/redfish/v1/", http.StatusUnauthorized)
+	return m
+}
+
+// NoSystem configures the server with a valid response for /redfish/v1
+func (m *RedfishMock) NoSystem() *RedfishMock {
+	m.rootService()
+	m.ResponseWithCode("/redfish/v1/Systems", `
+	{
+		"@odata.context": "/redfish/v1/$metadata#ComputerSystemCollection.ComputerSystemCollection",
+		"@odata.id": "/redfish/v1/Systems",
+		"@odata.type": "#ComputerSystemCollection.ComputerSystemCollection",
+		"Description": "Collection of Computer Systems",
+		"Members": [],
+		"Members@odata.count": 0,
+		"Name": "Computer System Collection"
+	}`, http.StatusOK)
+	return m
+}
+
+// Systems configures the server with a valid response for /redfish/v1/Systems
+func (m *RedfishMock) systems() *RedfishMock {
+	m.ResponseWithCode("/redfish/v1/Systems", `
+	{
+		"@odata.context": "/redfish/v1/$metadata#ComputerSystemCollection.ComputerSystemCollection",
+		"@odata.id": "/redfish/v1/Systems",
+		"@odata.type": "#ComputerSystemCollection.ComputerSystemCollection",
+		"Description": "Collection of Computer Systems",
+		"Members": [
+			{
+				"@odata.id": "/redfish/v1/Systems/System-1"
+			}
+		],
+		"Members@odata.count": 1,
+		"Name": "Computer System Collection"
+	}`, http.StatusOK)
+	return m
+}
+
+// System1 configures the server with a valid response for /redfish/v1/Systems/System-1
+func (m *RedfishMock) System1() *RedfishMock {
```

Comment:
See above. By having a general `System` method that accept a gofish.ComputerSystem object, this one will not be required

Reply:
Thanks!